test/box_test.dart

```diff
@@ -92,12 +99,243 @@ void main() {
       expect(fetchedItems[1], equals(null));
       expect(fetchedItems[2].text, equals("Two"));
     });
+  });
+
+  group("query", () {
```

Comment:
Specifically for queries, since there are so many different combinations, it would take too much time to use "standard" approach of creating a query, executing it and checking its result to cover 100 % query type/condition combinations (especially in Go, since there are currently 17 supported types...) 
In order to simplify it, you can define test-cases and just execute each of them using the same code. 
In it's most simple form a test case is a pair of `expectedQueryDescribeParamsResult` and `QueryCondition`. You pass this to a test "executor":
```dart
void testQuery(testCase) {
var query = box.query(testCase.conditions).build()
var actualDesc = query.describeParams()
expect(actualDesc, testCase.expectedDesc)
}
```

What Go query tests do on the line I've linked above does exactly that: call `testQueries()` with a huge list of test cases. They're a little more complex then just comparing description but that is actually the most important thing (as long as you test at least a couple of queries fully - which you already do.

Reply:
I've noticed you added these tests, thanks.